web/src/compose_banner.ts

```diff
@@ -97,5 +97,10 @@ export function show_stream_does_not_exist_error(stream_name: string): void {
     const $compose_banner_area = $("#compose_banners");
     $compose_banner_area.append(new_row);
     hide_compose_spinner();
-    $("#stream_message_recipient_stream").trigger("focus").trigger("select");
+
+    // TODO: Replace with compose_actions.open_compose_stream_dropup() when it is converted to ts.
+    if ($("#id_compose_select_stream").hasClass("open")) {
+        return;
+    }
+    $("#id_compose_select_stream button").trigger("click");
```

Comment:
I don't think we can have this module import `compose_actions.ts` in any case, because that'd generate an import cycle. 

The long term solution is almost certainly that we should be using stream IDs, not names, as the fundamental `compose_state` storage for streams (and probably also the `.value` for the dropdown list widget), which would lead to this error becoming impossible. So I think we can probably leave this as-is but change the TODO concept to be deleting this validator after migrating to stream IDs as how we keep track of the compose box recipient, with the name just used for display.

Reply:
Great point! Updated the comment.